src/jit/gtlist.h

```diff
@@ -93,6 +93,8 @@ GTNODE(SIMD_CHK         , "simdChk"      ,GenTreeBoundsChk   ,0,GTK_SPECIAL|GTK_
 
 GTNODE(ALLOCOBJ         , "allocObj"     ,GenTreeAllocObj    ,0,GTK_UNOP|GTK_EXOP)      // object allocator
 
+GTNODE(INIT_VAL         , "initVal"      ,GenTreeOp          ,0,GTK_UNOP)               // Initialization value for an initBlk
```

Comment:
Previously I was using a flag on the assignment/block op to indicate that it is an initBlk (vs. a copyBlk), but that causes problems with optimizations because it can replace a lclVar with a constant. We could simply handle TYP_STRUCT constant nodes, but that is messy because the initVal is actually really a byte constant expanded to the size of a struct. So this seems the best solution.

---
In reply to: [84719555](https://github.com/dotnet/coreclr/pull/7677#discussion_r84719555) [](ancestors = 84719555)